app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/MainActivity.kt

```diff
@@ -157,6 +171,43 @@ class MainActivity : BaseActivity(),
         viewModel.checkPostNotificationsPermission()
     }
 
+    @Composable
+    private fun ScheduleChangesStatistic() {
+        EventFahrplanTheme {
+            scheduleChangeStatisticsUiState.value?.let {
+                ChangesStatisticScreen(
+                    uiState = it,
+                    onConfirm = {
+                        AppRepository.updateScheduleChangesSeen(true)
+                        scheduleChangeStatisticsUiState.value = null
+                        openSessionChanges()
+                    },
+                    onDismiss = {
+                        AppRepository.updateScheduleChangesSeen(true)
+                        scheduleChangeStatisticsUiState.value = null
+                    },
+                )
+            }
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
```

Comment:
Instead of persisting the state, have you considered using a `StateFlow` to hold the state in the view model? That way it would survive orientation changes. And in case of process death the data seems to be reloaded via `MainViewModel.observeLoadScheduleState()`.

Reply:
Do you suggest change `mutableChangeStatisticsUiState = Channel<ChangeStatisticsUiState>()` to `mutableChangeStatisticsUiState = StateFlow<ChangeStatisticsUiState>()`? I decided to use a `Channel` so I can `sendOneTimeEvent` in order to launch the dialog only once.